From a3278a91e3883ac571b8c12f6159592f3e3f7b96 Mon Sep 17 00:00:00 2001
From: Wang Wensheng <wangwensheng4@huawei.com>
Date: Tue, 21 Feb 2023 03:48:52 +0000
Subject: [PATCH 03/53] [Huawei] mm/sharepool: Rename structure elements

Offering: HULK
hulk inclusion
category: feature
bugzilla: N/A

--------------------------------

sp_group_master.node_list => sp_group_master.group_head
sp_group_master.count => sp_group_master.group_num
sp_group.procs => sp_group.proc_head

Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
---
 mm/share_pool.c | 62 ++++++++++++++++++++++++-------------------------
 1 file changed, 31 insertions(+), 31 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 74871f4f30ac..a3ae16f06fd6 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -168,7 +168,7 @@ struct sp_group {
 	/* number of process in this group */
 	int		 proc_num;
 	/* list head of processes (sp_group_node, each represents a process) */
-	struct list_head procs;
+	struct list_head proc_head;
 	/* list head of sp_area. it is protected by spin_lock sp_area_lock */
 	struct list_head spa_list;
 	/* group statistics */
@@ -189,11 +189,11 @@ struct sp_group_master {
 	pid_t tgid;
 	/*
 	 * number of sp groups the process belongs to,
-	 * a.k.a the number of sp_node in node_list
+	 * a.k.a the number of sp_node in group_head
 	 */
-	unsigned int count;
+	unsigned int group_num;
 	/* list head of sp_node */
-	struct list_head node_list;
+	struct list_head group_head;
 	struct mm_struct *mm;
 	/*
 	 * Used to apply for the shared pool memory of the current process.
@@ -209,12 +209,12 @@ struct sp_group_master {
  * each instance represents an sp group the process belongs to
  * sp_group_master    : sp_group_node   = 1 : N
  * sp_group_node->spg : sp_group        = 1 : 1
- * sp_group_node      : sp_group->procs = N : 1
+ * sp_group_node      : sp_group->proc_head = N : 1
  */
 struct sp_group_node {
-	/* list node in sp_group->procs */
+	/* list node in sp_group->proc_head */
 	struct list_head proc_node;
-	/* list node in sp_group_maseter->node_list */
+	/* list node in sp_group_maseter->group_head */
 	struct list_head group_node;
 	struct sp_group_master *master;
 	struct sp_group *spg;
@@ -449,7 +449,7 @@ static int sp_mapping_group_setup(struct mm_struct *mm, struct sp_group *spg)
 	local_dvpp_mapping = mm->sp_group_master->local->mapping[SP_MAPPING_DVPP];
 	spg_dvpp_mapping = spg->mapping[SP_MAPPING_DVPP];
 
-	if (!list_empty(&spg->procs)) {
+	if (!list_empty(&spg->proc_head)) {
 		/*
 		 * Don't return an error when the mappings' address range conflict.
 		 * As long as the mapping is unused, we can drop the empty mapping.
@@ -562,8 +562,8 @@ static int sp_init_group_master_locked(struct task_struct *tsk, struct mm_struct
 	if (!master)
 		return -ENOMEM;
 
-	INIT_LIST_HEAD(&master->node_list);
-	master->count = 0;
+	INIT_LIST_HEAD(&master->group_head);
+	master->group_num = 0;
 	master->mm = mm;
 	master->tgid = tsk->tgid;
 	get_task_comm(master->comm, current);
@@ -814,7 +814,7 @@ static struct sp_group_node *find_spg_node_by_spg(struct mm_struct *mm,
 {
 	struct sp_group_node *spg_node;
 
-	list_for_each_entry(spg_node, &mm->sp_group_master->node_list, group_node) {
+	list_for_each_entry(spg_node, &mm->sp_group_master->group_head, group_node) {
 		if (spg_node->spg == spg)
 			return spg_node;
 	}
@@ -954,7 +954,7 @@ static bool is_process_in_group(struct sp_group *spg,
 {
 	struct sp_group_node *spg_node;
 
-	list_for_each_entry(spg_node, &spg->procs, proc_node)
+	list_for_each_entry(spg_node, &spg->proc_head, proc_node)
 		if (spg_node->master->mm == mm)
 			return true;
 
@@ -1049,7 +1049,7 @@ int mg_sp_group_id_by_pid(int tgid, int *spg_ids, int *num)
 	 * implementation for convenience and is not attempt to bother
 	 * the user.
 	 */
-	real_count = master->count - 1;
+	real_count = master->group_num - 1;
 	if (real_count <= 0) {
 		ret = -ENODEV;
 		goto out_up_read;
@@ -1060,7 +1060,7 @@ int mg_sp_group_id_by_pid(int tgid, int *spg_ids, int *num)
 	}
 	*num = real_count;
 
-	list_for_each_entry(node, &master->node_list, group_node) {
+	list_for_each_entry(node, &master->group_head, group_node) {
 		if (is_local_group(node->spg->id))
 			continue;
 		*(spg_ids++) = node->spg->id;
@@ -1085,7 +1085,7 @@ static void sp_group_init(struct sp_group *spg, int spg_id)
 	spg->proc_num = 0;
 	atomic_set(&spg->use_count, 1);
 	atomic_set(&spg->spa_num, 0);
-	INIT_LIST_HEAD(&spg->procs);
+	INIT_LIST_HEAD(&spg->proc_head);
 	INIT_LIST_HEAD(&spg->spa_list);
 	INIT_LIST_HEAD(&spg->mnode);
 	init_rwsem(&spg->rw_lock);
@@ -1224,7 +1224,7 @@ static int mm_add_group_init(struct task_struct *tsk, struct mm_struct *mm,
 		}
 
 		master = mm->sp_group_master;
-		if (master->count == MAX_GROUP_FOR_TASK) {
+		if (master->group_num == MAX_GROUP_FOR_TASK) {
 			pr_err("task reaches max group num\n");
 			return -ENOSPC;
 		}
@@ -1251,8 +1251,8 @@ static struct sp_group_node *create_spg_node(struct mm_struct *mm,
 	spg_node->prot = prot;
 	meminfo_init(&spg_node->meminfo);
 
-	list_add_tail(&spg_node->group_node, &master->node_list);
-	master->count++;
+	list_add_tail(&spg_node->group_node, &master->group_head);
+	master->group_num++;
 
 	return spg_node;
 }
@@ -1266,7 +1266,7 @@ static int insert_spg_node(struct sp_group *spg, struct sp_group_node *node)
 	}
 
 	spg->proc_num++;
-	list_add_tail(&node->proc_node, &spg->procs);
+	list_add_tail(&node->proc_node, &spg->proc_head);
 
 	return 0;
 }
@@ -1285,7 +1285,7 @@ static void free_spg_node(struct mm_struct *mm, struct sp_group *spg,
 	struct sp_group_master *master = mm->sp_group_master;
 
 	list_del(&spg_node->group_node);
-	master->count--;
+	master->group_num--;
 
 	kfree(spg_node);
 }
@@ -1942,7 +1942,7 @@ static void __sp_free(struct sp_group *spg, unsigned long addr,
 	struct mm_struct *mm;
 	struct sp_group_node *spg_node = NULL;
 
-	list_for_each_entry(spg_node, &spg->procs, proc_node) {
+	list_for_each_entry(spg_node, &spg->proc_head, proc_node) {
 		mm = spg_node->master->mm;
 		if (mm == stop)
 			break;
@@ -2375,7 +2375,7 @@ static int sp_alloc_mmap_populate(struct sp_area *spa,
 		__nodemask = *nodemask;
 
 	/* create mapping for each process in the group */
-	list_for_each_entry(spg_node, &spa->spg->procs, proc_node) {
+	list_for_each_entry(spg_node, &spa->spg->proc_head, proc_node) {
 		mm = spg_node->master->mm;
 		mmap_ret = __sp_alloc_mmap_populate(mm, spa, spg_node, ac, &__nodemask);
 		if (mmap_ret) {
@@ -2399,7 +2399,7 @@ static int sp_alloc_mmap_populate(struct sp_area *spa,
 
 unmap:
 	/* use the next mm in proc list as end mark */
-	if (!list_is_last(&spg_node->proc_node, &spa->spg->procs))
+	if (!list_is_last(&spg_node->proc_node, &spa->spg->proc_head))
 		end_mm = list_next_entry(spg_node, proc_node)->master->mm;
 	sp_alloc_unmap(end_mm, spa, spg_node);
 
@@ -2643,7 +2643,7 @@ static void *sp_make_share_kva_to_spg(struct sp_k2u_context *kc, struct sp_group
 	}
 
 	spa->kva = kc->kva_aligned;
-	list_for_each_entry(spg_node, &spg->procs, proc_node) {
+	list_for_each_entry(spg_node, &spg->proc_head, proc_node) {
 		mm = spg_node->master->mm;
 		kc->state = K2U_NORMAL;
 		ret_addr = sp_remap_kva_to_vma(spa, mm, spg_node->prot, kc);
@@ -3507,7 +3507,7 @@ static void get_process_sp_res(struct sp_group_master *master,
 	*sp_res_out = 0;
 	*sp_res_nsize_out = 0;
 
-	list_for_each_entry(spg_node, &master->node_list, group_node) {
+	list_for_each_entry(spg_node, &master->group_head, group_node) {
 		spg = spg_node->spg;
 		*sp_res_out += meminfo_alloc_sum_byKB(&spg->meminfo);
 		*sp_res_nsize_out += byte2kb(atomic64_read(&spg->meminfo.alloc_nsize));
@@ -3586,7 +3586,7 @@ int proc_sp_group_state(struct seq_file *m, struct pid_namespace *ns,
 	seq_printf(m, "%-8s %-9s %-9s %-9s %-4s\n",
 			"Group_ID", "SP_ALLOC", "SP_K2U", "SP_RES", "PROT");
 
-	list_for_each_entry(spg_node, &master->node_list, group_node) {
+	list_for_each_entry(spg_node, &master->group_head, group_node) {
 		seq_printf(m, "%-8d %-9ld %-9ld %-9ld ",
 				spg_node->spg->id,
 				meminfo_alloc_sum_byKB(&spg_node->meminfo),
@@ -3785,7 +3785,7 @@ static int proc_usage_by_group(int id, void *p, void *data)
 	unsigned long anon, file, shmem, total_rss;
 
 	down_read(&spg->rw_lock);
-	list_for_each_entry(spg_node, &spg->procs, proc_node) {
+	list_for_each_entry(spg_node, &spg->proc_head, proc_node) {
 		master = spg_node->master;
 		mm = master->mm;
 		tgid = master->tgid;
@@ -4029,19 +4029,19 @@ int sp_group_exit(void)
 		return 0;
 	}
 
-	list_for_each_entry_safe(spg_node, tmp, &master->node_list, group_node) {
+	list_for_each_entry_safe(spg_node, tmp, &master->group_head, group_node) {
 		spg = spg_node->spg;
 
 		down_write(&spg->rw_lock);
 		/* a dead group should NOT be reactive again */
-		if (spg_valid(spg) && list_is_singular(&spg->procs))
+		if (spg_valid(spg) && list_is_singular(&spg->proc_head))
 			spg->is_alive = false;
 		delete_spg_node(spg, spg_node);
 		up_write(&spg->rw_lock);
 	}
 
 	/* match with get_task_mm() in sp_group_add_task() */
-	if (atomic_sub_and_test(master->count, &mm->mm_users)) {
+	if (atomic_sub_and_test(master->group_num, &mm->mm_users)) {
 		up_write(&sp_group_sem);
 		WARN(1, "Invalid user counting\n");
 		return 1;
@@ -4087,7 +4087,7 @@ void sp_group_post_exit(struct mm_struct *mm)
 			byte2kb(alloc_size), byte2kb(k2u_size));
 
 	down_write(&sp_group_sem);
-	list_for_each_entry_safe(spg_node, tmp, &master->node_list, group_node) {
+	list_for_each_entry_safe(spg_node, tmp, &master->group_head, group_node) {
 		spg = spg_node->spg;
 		/* match with refcount inc in sp_group_add_task */
 		if (atomic_dec_and_test(&spg->use_count))
-- 
2.17.1

